site_scons/buildutils.py

```diff
@@ -1,118 +1,280 @@
+
+    Intended to be used with the LevelAdapter class to allow the ``print_level``
+    keyword argument to be added to Logger method calls (``.info()`` and others). The
+    ``print_level`` Boolean value is used to determine whether or not the level of the
+    logging message should be printed. By default, the level is shown. Example::
+
+       >>> logger.info("Message", print_level=False)
+       Message
+       >>> logger.info("Message")
+       INFO: Message
+    """
+
+    def format(self, record: logging.LogRecord) -> str:
+        if record.exc_info or record.exc_text:
+            raise ValueError("This formatter does not support exceptions")
+        elif record.stack_info:
+            raise ValueError("This formatter does not support stack traces")
+
+        no_level_style = "{message}"
+        level_style = "{levelname}: " + no_level_style
+        record.message = record.getMessage()
+        if getattr(record, "print_level", True):
+            s = level_style.format(**record.__dict__)
+        else:
+            s = no_level_style.format(**record.__dict__)
+        return s
+
+
+# Modified from https://stackoverflow.com/a/36338212
+class LevelFilter(logging.Filter):
+    """Filter out log messages above or below preset cutoffs.
+
+    Log levels in Python correspond to integers, with the lowest, DEBUG, set to
+    10 and the highest, CRITICAL, set to 50. This filter causes a log handler to
+    reject messages that are above or below numerical cutoffs. Example::
+
+    >>> # Handles log levels from debug up to, but not including, error
+    >>> handler.addFilter(LevelFilter(logging.DEBUG, logging.ERROR))
+    >>> # Handles log levels from warning up to and including critical
+    >>> handler.addFilter(LevelFilter(logging.WARNING, logging.CRITICAL+1))
+    """
+
+    def __init__(self, low: int, high: int) -> None:
+        self._low = low
+        self._high = high
+        super().__init__()
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        if self._low <= record.levelno < self._high:
+            return True
+        return False
+
+
+logging.setLogRecordFactory(BraceLogRecord)
+logger = logging.getLogger("cantera")
+logger.setLevel(logging.INFO)
+f = OutputFormatter()
+stdout_handler = logging.StreamHandler(sys.stdout)
+stdout_handler.setFormatter(f)
+stdout_handler.addFilter(LevelFilter(logging.DEBUG, logging.ERROR))
+logger.addHandler(stdout_handler)
+
+stderr_handler = logging.StreamHandler(sys.stderr)
+stderr_handler.setFormatter(f)
+stderr_handler.addFilter(LevelFilter(logging.ERROR, logging.CRITICAL + 1))
+logger.addHandler(stderr_handler)
+
+logger = LevelAdapter(logger)
+
+
+if sys.version_info[:2] < (3, 6):
+    logger.error(
+        "Cantera must be built using Python 3.6 or higher. If you are using "
+        "a system install of SCons, you can invoke SCons by writing\n"
+        "    python3 `which scons`\n"
+        "followed by any desired options."
+    )
+    sys.exit(1)
```

Comment:
Given the `SyntaxError`-generating f-string in `SConstruct`, this is no longer necessary, right?